model/textparse/openmetricsparse.go

```diff
@@ -213,7 +213,9 @@ func (p *OpenMetricsParser) Comment() []byte {
 
 // Labels writes the labels of the current sample into the passed labels.
 func (p *OpenMetricsParser) Labels(l *labels.Labels) {
-	s := yoloString(p.series)
+	// Defensive copy in case the following keeps a reference.
+	// See https://github.com/prometheus/prometheus/issues/16490
+	s := string(p.series)
```

Comment:
easier/safer than to make sure the consumers don't keep a reference

Reply:
This will be a performance regression for `stringlabels`.
I think if the `Add` below was `AddUnsafeBytes` then we would get the best of both worlds.
We'd also need to rewrite `unreplace` to work on `[]byte`.

Reply:
also, in Prometheus, with the scrape cache, we shouldn't go down this path that much (https://github.com/prometheus/prometheus/blob/1d9dfde9894f0c82b809b70aa5f9cecab25b8fea/scrape/scrape.go#L1728-L1739)